ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -162,9 +160,7 @@ let disable ~__context ~self =
       assert_operation_host_target_is_localhost ~__context ~host;
       assert_cluster_host_has_no_attached_sr_which_requires_cluster_stack ~__context ~self;
 
-      let pool = Helpers.get_pool ~__context in
-      if Db.Pool.get_ha_enabled ~__context ~self:pool then
-        Db.Pool.set_ha_cluster_stack ~__context ~self:pool ~value:"xhad";
+      set_ha_cluster_stack_if_ha_enabled ~__context ~value:"xhad";
```

Comment:
Hmm, maybe we don't need to set / unset this pool field if we only enable / disable one cluster_host?

Reply:
Perhaps it would make more sense to check across all cluster hosts?

Reply:
@edwintorok Why do we need to switch to xhad if we only disable one cluster host?

Reply:
You are right, the cluster stack should not be changed at all in Xapi_cluster_host

Reply:
Just simply creating a cluster should be enough to make it incompatible with xhad, but like I said in my other comment, the code should be changed in Cluster_constraints.ml too.